Re exec only when asked

spraints opened the pull request:
This is a rebased copy of #53. We did have a problem where the binstub for resqued wasn't working during a deploy, and that ended up stopping all of our workers instead of failing gracefully.

spraints committed:
resqued/exec_on_hup -> resqued/replace_master

spraints committed:
Re-exec on USR1, not HUP

Instead of re-execing the master process on every HUP, only do it when a
USR1 signal is received. I'm a little concerned that there will be some
long-run problem with re-execing on every restart. This is me working
out my paranoia in a more productive (?) way.

spraints committed:
Revise Documentation todo

spraints committed:
Document MasterState

erichhsun reviewed (approved):
:+1:

I think the `README.md` will also need to be updated:

https://github.com/spraints/resqued/blob/7d8394cff76cfc8c39e3dc9086a05d5419834924/README.md?plain=1#L60

Maybe it's also worth providing an explicit usage of SIGHUP, i.e. using it in systemctl's `ExecReload` for deploys to avoid issues like a code change breaking the master and therefore crashing existing workers.

spraints committed:
Fix CI badge

spraints committed:
Update readme

Fix the description of how resqued restarts.

Replace the upstart example with a systemd example.

spraints commented on lib/resqued/master_state.rb:
```diff
@@ -1,11 +1,14 @@
+    # initial values.
     def init(options)
       @config_paths = options.fetch(:config_paths)
       @exec_on_hup  = options.fetch(:exec_on_hup, false)
```
Good catch! I am going to leave it in place for now so that if we end up rolling back in prod, we'll preserve the setting in the previous version of resqued. But I'll follow up later to remove this setting.

erichhsun reviewed (approved):
✨
